src/presets/firebase.ts

```diff
@@ -51,9 +51,14 @@ async function writeRoutes(nitro: Nitro) {
 
   const _require = createRequire(import.meta.url);
 
-  const jsons = await globby(
+  const allJsons = await globby(
     join(nitro.options.output.serverDir, "node_modules/**/package.json")
   );
+
+  // fsevents is a macOS specific package that breaks in google cloud build, so we filter-it ot
+  // see: https://github.com/unjs/nitro/issues/1367
+  const jsons = allJsons.filter((json) => !json.includes("fsevents"));
```

Comment:
Would it make more sense to check for something more specific like `"name": "fsevents"`

Reply:
@posva granted, `jsons` is not a great name for this const, but if you follow the code, it comes from globby:

```ts
const jsons = await globby(
    join(nitro.options.output.serverDir, "node_modules/**/package.json")
  );
```

globby is going to return an array of file paths like:

```ts
[
  "/Users/USER_NAME/PROJECT_PATH/.output/server/node_modules/anymatch/package.json",
  "/Users/USER_NAME/PROJECT_PATH/.output/server/node_modules/binary-extensions/package.json",
  "/Users/USER_NAME/PROJECT_PATH/.output/server/node_modules/buffer-equal-constant-time/package.json",
  "/Users/USER_NAME/PROJECT_PATH/.output/server/node_modules/braces/package.json",
  "/Users/USER_NAME/PROJECT_PATH/.output/server/node_modules/buffer-from/package.json",
  "/Users/USER_NAME/PROJECT_PATH/.output/server/node_modules/chokidar/package.json",
  "/Users/USER_NAME/PROJECT_PATH/.output/server/node_modules/fsevents/package.json",
  // ... etc
]
```

So this is why we're using an `includes` function to strip out any globby paths that have a match for fsevents.

We could use some regex or a more specific string in includes here to make sure that for instance if a user's home dir on a mac is `/Users/fsevents` because thats what they used when they set up their mac...?

I'm not sure how specific we want to be - seems like a very rare case...

I guess we could check for `"fsevents/package.json"` and filter those out to be more specific?

Reply:
Or we could move the check down into the reduce call after the require:

```ts
const dependencies = jsons.reduce((obj, packageJson) => {
  const dirname = packageJson.slice(prefixLength, -suffixLength);
  if (!dirname.includes("node_modules")) {
    const packageJsonObject = _require(packageJson)
    if (packageJsonObject.name !== "fsevents") {
      obj[dirname] = _require(packageJson).version;
    }
  }
  return obj;
}, {} as Record<string, string>);
```

What do you think @Hebilicious ?

Reply:
In that case you can exclude directly using the options of the globby function then